src/app/course/course-question-snippet/course-question-snippet.component.ts

```diff
@@ -169,9 +169,29 @@ export class CourseQuestionSnippetComponent implements OnInit {
             }).subscribe()
         } else {
             this.router.navigate(
-                ['../' , this.eventId, 'problem', uqj.question.id, 'edit']
-                ,{relativeTo: this.route}
+                ['../', this.eventId, 'problem', uqj.question.id, 'edit']
+                , {relativeTo: this.route}
             ).then()
         }
     }
+
+    /**
+     * Opens the dialog service based on the template passed
+     * @param content - the template to be used
+     * @param finished - the boolean used to alter label text
+     * @param labelText - the string used to alternate label text
+     */
+    openRemoveQuestionInEventDialog(
+        content: PolymorpheusContent<TuiDialogContext>,
+        finished: boolean,
+        labelText: string
```

Comment:
The logic of `labelText` is split in 2 different files. This is a high-level architecture issue. Remove the `labelText` from the arguments of this method and calculate it inside.

To improve this further, you can avoid logic in the HTML file by passing the `event` to this method instead of `finished` and then calculating the `finished` inside.